content/markdown/04-intro-to-stats/04-09-generalized-regression.Rmd

```diff
@@ -0,0 +1,535 @@
+
+### Link Function 
+
+This is the heart of what makes a model a GLM. This component essentially allows us to relax the assumption of normality of our response variable, by allowing other distributions, and then defining some function of the response variable that in fact DOES vary linearly with the predictor variables. The math behind how this works is both incredibly interesting, and requires some knowledge of linear algebra, so we will leave the interested reader with an encouragement to take a linear algebra class if they have room in their timetable. 
+
+Suffice it to say here that a link function allows us to take some non-linear response data, and "transform" it such that it now varies linearly with the predictor variable(s). 
+
+### Error Distribution
+
+The error distribution we define will in turn define which link function we choose. Determining what error distribution we need to choose can be crudely but often satisfactorily chosen by plotting the response variable, but a more nuanced approach is possible. See <a href="https://r.qcbs.ca/workshop07/book-en/choose-an-error-distribution.html>this tutorial</a> from the Quebec Centre for Biodiversity Science for an example. 
+
+### Species Richness Example - I
+
+Back to our example stated earlier regarding if the relationship between biomass and species richness changes with soil pH. The data for this example can be found <a href="https://exeter-data-analytics.github.io/StatModelling/_data/species.csv">here</a>. If you want to download the data and follow along, be sure to check out the <a href="../01-basicr/01-06-workflow.html">Workflow section</a>. 
+
+Since we already have our hypothesis stated (and therefore cannot p-hack ourselves), we can proceed to plot our data. Let's first inspect it briefly:
+
+```{r}
+library(tidyverse)
+# uncomment and edit the lines below if you need to read in the data 
+# library(here)
+# df <- readr::read_csv(here("./data/species.csv"))
+
+# quick summary of our data
+dplyr::glimpse(df)
+head(df)
+```
+
+Ok so we can see what types are data are. It's important here to not be tricked by R!! It has read in the data automatically with the <code>Species</code> column as numeric type, but it's in fact integer values. This doens't matter right now, but it will matter when we need to choose an error distribution. 
+
+#### Step 1 - Plot Data
+
+Let's plot our data to see about what we're dealing with. 
+
+```{r}
+library(ggthemes)
+
+ggplot(data = df) + 
+  geom_point(mapping = aes(x = Biomass, y = Species)) +
+  ggthemes::theme_base()
+```
+
+It even looks like there's a linear decrease in richness as biomass increases. So why can't we just assume a normal distribution here? Well, let's take a look at it. 
+
+#### An Aside - Motivation for Link Functions
```

Comment:
I think this motivation is more complex than necessary. Some variables are just not normally distributed